float32 distance
float32 cur_height
uint16 height
uint8 semantic